pip/build_manylinux.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Is to be executed in the Manylinux Docker container.
+# https://github.com/pypa/manylinux
+
+# Generates the manylinux .whl file.
+# Input:
+#   "/io" is mapped to this repo dir "qsharp" (from YAML, from repo dir  "docker run -v `pwd`:/io").
+#   Environment (from YAML
+#   `docker run --env VER_MAJOR=$(Ver.Major) --env VER_MINOR=$(Ver.Minor) --env VER_PATCH=$(Ver.Patch) \
+#   --env ARCH=x86_64`):
+#       VER_MAJOR: $(Ver.Major)
+#       VER_MINOR: $(Ver.Minor)
+#       VER_PATCH: $(Ver.Patch)
+#       ARCH: Architecture: x86_64 or aarch64, etc.
+
+set -x -e
+
+# Install Rust:
+curl --proto '=https' --tlsv1.2 -sS https://sh.rustup.rs > rustup-init.sh
+chmod +x rustup-init.sh
+./rustup-init.sh -y
+source "$HOME/.cargo/env"
+
+# Install "Node.js":
+# https://stackoverflow.com/a/73385596/6362941
+curl -sL https://rpm.nodesource.com/setup_18.x | bash -
```

Comment:
This isn't the official Node.js url - it seems like a third party company that happens to have a setup script for Node.js... Is it OK to take a dependency on them?

I'd refer to their install instructions on the website https://nodejs.org/en/download

Reply:
Now that I think about it, is Node.js even needed? It shouldn't be, for the platform-specific wheel (qsharp). 

Node.js is only required for qsharp_jupyterlab but that one is not platform specific, so we probably don't need to build it in a manylinux container.

Reply:
That will simplify a number of things... I will experiment.

Reply:
Interesting... @billti, @swernli, @idavis, @samarsha, @minestarks, please share your thoughts.
1. "build.py" UNconditionally calls `check_prereqs()`. 
`check_prereqs()` forces to install `nodejs`.

2. `nodejs` is a pre-req for `npm`.
`npm` is used for `--check` and `--check` is the default.

So, in the Docker containers we 
* either always install `nodejs`, 
* or 
  * disable the call from "build.py" to `check_prereqs()` (or make that call "_picky_" by passing args to it)
  * AND build the .whl files with `--no-check` (optionally running `--check` before the container, AND {
    * we either enable running `--check` alone because currently if nothing is passed in addition to `--check` then _everything_ is built
    * OR running `--check` with something else, most likely `--pip`, such that `--pip` is built before and inside of the container}).
  
I personally feel that it is reasonable for the call from "build.py" to `check_prereqs()` to be "picky" (which is convenient for different use cases but is an extra maintenance headache), such that `check_prereqs()` requires the really required things only (which are different depending on what "build.py" builds). E.g. 
* for "build.py --jupyterlab" the nodejs is required by `check_prereqs()`
* but for "build.py --pip" the nodejs is NOT required by `check_prereqs()`.

After that the `--check` seems reasonable to be runnable as a separate thing.

Then we can avoid installing in docker containers the nodejs from a third-party (I failed to find better ways to install the nodejs of version required by our build scripts (16.17+) in containers).

Reply:
Great point. In my opinion, in Publish tasks, we can _always_ pass `--no-check` and even `--no-tests` to build.py. In the same spirit, we could define a new `--no-check-prereqs` flag which skips `check_prereqs()`, and pass that in as well.

My reasoning is that checking and testing are already done in the CI build for every PR. There is not really a reason to repeat those steps in the publishing action as well.

Reply:
Installing nodejs has been removed from the Docker containers.
The `--no-check-prereqs` has been added.
The `--no-check` and `--no-tests` have been added to a number of calls.

Other changes are planned to be done when working on ARM64 for Windows and Mac.